src/core/Akka.Streams.Tests/Dsl/AsyncEnumerableSpec.cs

```diff
@@ -230,6 +225,47 @@ public async Task AsyncEnumerableSource_Must_Cancel_Running_Source_If_Downstream
             // The cancellation token inside the IAsyncEnumerable should be cancelled
             await WithinAsync(3.Seconds(), async () => latch.Value);
         }
+        
+        /// <summary>
+        /// Reproduction for https://github.com/akkadotnet/akka.net/issues/6280
+        /// </summary>
+        [Fact]
+        public async Task AsyncEnumerableSource_BugFix6280()
+        {
+            async IAsyncEnumerable<int> GenerateInts()
+            {
+                foreach (var i in Enumerable.Range(0, 100))
+                {
+                    if (i > 50)
+                        await Task.Delay(1000);
```

Comment:
This is designed to force the `ValueTask` returned from `IAsyncEnumerator.MoveNextAsync()` to be still running at the moment we attempt to dispose the `IAsyncEnumerator`.